src/server/ua_services_subscription.c

```diff
@@ -185,13 +190,18 @@ setMonitoredItemSettings(UA_Server *server, UA_MonitoredItem *mon,
         mon->samplingInterval = server->config.samplingIntervalLimits.min;
 
     /* Filter */
-    if(params->filter.encoding != UA_EXTENSIONOBJECT_DECODED ||
-       params->filter.content.decoded.type != &UA_TYPES[UA_TYPES_DATACHANGEFILTER]) {
+    if(params->filter.encoding != UA_EXTENSIONOBJECT_DECODED) {
         /* Default: Trigger only on the value and the statuscode */
-        mon->trigger = UA_DATACHANGETRIGGER_STATUSVALUE;
+        mon->filter.trigger = UA_DATACHANGETRIGGER_STATUSVALUE;
+    } else if (params->filter.content.decoded.type == &UA_TYPES[UA_TYPES_EVENTFILTER]) {
+        mon->filter.eventFilter = UA_EventFilter_new();
+        if (!mon->filter.eventFilter) {
+            return;
```

Comment:
We probably should show an error message here. Something like "Out of memory creating event filter"